@here/harp-theme-tools/src/cli-build-theme.ts

```diff
@@ -40,25 +41,65 @@ if (typeof cliOptions.in !== "string" || cliOptions.in.length === 0) {
 
 process.chdir(cliOptions.chdir);
 
-const inputFiles = glob.sync(cliOptions.in);
+const inFileNames = glob.sync(cliOptions.in);
 
-if (inputFiles.length === 0) {
+if (inFileNames.length === 0) {
     console.error("No input files found");
     program.outputHelp();
     process.exit(1);
 }
 
-inputFiles.forEach(file => {
-    ThemeLoader.load(file, {
+inFileNames.forEach(inFileName => {
+    ThemeLoader.load(inFileName, {
         resolveResourceUris: false,
         resolveIncludeUris: true
     }).then(theme => {
         theme.url = undefined;
-        const filename = `${cliOptions.out}/${path.basename(file)}`;
-        console.log(`Writing ${filename}`);
         const json = cliOptions.minify
             ? JSON.stringify(theme)
             : JSON.stringify(theme, undefined, 4);
-        fs.writeFileSync(filename, json);
+
+        const unifiedDefinitionsFileName =
+            path.dirname(inFileName) +
+            path.sep +
+            "unified-definitions" +
+            path.sep +
+            path.basename(inFileName);
+
+        // If unified definitions file exists then we split a theme to unified with native and
+        // non-unified with native parts. We need such split to explain users which parts of
+        // the theme will stay backward compatible and which ones will not:
+        if (fs.existsSync(unifiedDefinitionsFileName)) {
+            // Private (non-unified) part:
+            const outFileBaseNamePrivate = `${path.basename(inFileName, "json")}private.json`;
+            const outFileNamePrivate = cliOptions.out + path.sep + outFileBaseNamePrivate;
+            console.log(`Writing ${outFileNamePrivate}`);
+            fs.writeFileSync(outFileNamePrivate, json);
+
+            // Public (unified) part:
+            const themePublic: FlatTheme = {
+                extends: [outFileBaseNamePrivate],
+                definitions: {},
+                url: undefined
```

Comment:
JSON.stringify omits all the properties with undefined values. No need to include ```url``` for the time being